inbm/dispatcher-agent/tests/unit/source/test_ubuntu_source_cmd.py

```diff
@@ -274,18 +282,52 @@ def test_successfully_remove_gpg_key_and_source_list(
         parameters = ApplicationRemoveSourceParameters(
             gpg_key_name="example_source.gpg", file_name="example_source.list"
         )
-        command = UbuntuApplicationSourceManager()
+        broker = MockDispatcherBroker.build_mock_dispatcher_broker()
+        command = UbuntuApplicationSourceManager(broker)
         try:
             command.remove(parameters)
         except SourceError:
             self.fail("Remove GPG key raised DispatcherException unexpectedly!")
 
+
+    @patch("dispatcher.source.ubuntu_source_manager.verify_source", side_effect=DispatcherException('error'))
+    def test_failed_add_gpg_key_method(self, mock_verify_source):
+        parameters = ApplicationAddSourceParameters(
+            file_name="intel-gpu-jammy.list",
+            sources="deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main",
+            gpg_key_uri="https://dl-ssl.google.com/linux/linux_signing_key.pub",
+            gpg_key_name="name"
+        )
+        broker = MockDispatcherBroker.build_mock_dispatcher_broker()
+        command = UbuntuApplicationSourceManager(broker)
+        try:
+            command.add(parameters)
```

Comment:
@tsirlapu this would pass even if there wasn't an exception--need to do something like this:

```
with pytest.raises(SourceError) as ex:
            command.remove(parameters)
assert str(ex.value) == "(whatever you expect the exception to be)"
```

and drop the try/except block